lib/trento/infrastructure/commanded/event_handlers/database_deregistration_event_handler.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Trento.Infrastructure.Commanded.EventHandlers.DatabaseDeregistrationEventHandler do
+  @moduledoc """
+  This event handler is responsible to forward deregistration commands to the SAP systems
+  related to a deregistered database
+  """
+
+  use Commanded.Event.Handler,
+    application: Trento.Commanded,
+    name: "database_deregistration_event_handler"
+
+  alias Trento.Databases.Events.DatabaseDeregistered
+  alias Trento.Repo
+  alias Trento.SapSystems.Commands.DeregisterSapSystem
+  alias Trento.SapSystems.Projections.DatabaseReadModel
+
+  import Ecto.Query, only: [from: 2]
+
+  require Logger
+
+  def handle(
+        %DatabaseDeregistered{database_id: database_id, deregistered_at: deregistered_at},
+        _metadata
+      ) do
+    database =
+      Repo.one!(
+        from(d in DatabaseReadModel,
+          where: d.id == ^database_id,
+          preload: [:sap_systems],
+          select: [:id]
+        )
+      )
+
+    for %{id: sap_system_id} <- database.sap_systems do
+      Logger.info(
+        "Deregistering sap system: #{sap_system_id} attached to database: #{database_id}"
```

Comment:
I think here the best for logging would be to use the `sid` rather than the id.
It should be pretty simple to get them in the previous query.
Anyway, we can change later on if we want as well.

PD: i would remove the `:` from the logging message. They look a bit disruptive

Reply:
done